velox/exec/HashProbe.cpp

```diff
@@ -355,17 +356,31 @@ void HashProbe::fillOutput(vector_size_t size) {
       output_);
 }
 
-RowVectorPtr HashProbe::getNonMatchingOutputForRightJoin() {
-  if (!lastRightJoinProbe_) {
+RowVectorPtr HashProbe::getBuildSideOutput() {
+  if (!(isRightJoin(joinType_) || isFullJoin(joinType_) ||
```

Comment:
Maybe add a CHECK for not these cases instead of returning nullptr, especially if you don't expect to call this function in those cases.